utils/use_require.js

```diff
@@ -38,46 +40,95 @@ const no_transform_files = new Set([
 
 no_copy_files.forEach((file) => no_transform_files.add(file));
 
+// util.promisify requires Node.js 8.x, so we have our own
+function promisify(original) {
+    return function () {
+        let obj = this;
+        let args = Array.prototype.slice.call(arguments);
+        return new Promise((resolve, reject) => {
+            original.apply(obj, args.concat((err, value) => {
+                if (err) return reject(err);
+                resolve(value);
+            }));
+        });
+    }
+}
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+const readdir = promisify(fs.readdir);
+const lstat = promisify(fs.lstat);
+
+const copy = promisify(fse.copy);
+const unlink = promisify(fse.unlink);
+const ensureDir = promisify(fse.ensureDir);
+const rmdir = promisify(fse.rmdir);
+
+const babelTransformFile = promisify(babel.transformFile);
+
 // walkDir *recursively* walks directories trees,
 // calling the callback for all normal files found.
 var walkDir = function (base_path, cb, filter) {
-    fs.readdir(base_path, (err, files) => {
-        if (err) throw err;
-
-        files.map((filename) => path.join(base_path, filename)).forEach((filepath) => {
-            fs.lstat(filepath, (err, stats) => {
-                if (err) throw err;
-
+    return readdir(base_path)
+    .then(files => {
+        let paths = files.map(filename => path.join(base_path, filename));
+        return Promise.all(paths.map((filepath) => {
+            return lstat(filepath)
+            .then(stats => {
                 if (filter !== undefined && !filter(filepath, stats)) return;
 
                 if (stats.isSymbolicLink()) return;
-                if (stats.isFile()) cb(filepath);
-                if (stats.isDirectory()) walkDir(filepath, cb, filter);
+                if (stats.isFile()) return cb(filepath);
+                if (stats.isDirectory()) return walkDir(filepath, cb, filter);
             });
-        });
+        }));
     });
 };
 
-var transform_html = function (new_script) {
+var transform_html = function (legacy_scripts) {
     // write out the modified vnc.html file that works with the bundle
     var src_html_path = path.resolve(__dirname, '..', 'vnc.html');
     var out_html_path = path.resolve(paths.out_dir_base, 'vnc.html');
-    fs.readFile(src_html_path, (err, contents_raw) => {
-        if (err) { throw err; }
-
+    return readFile(src_html_path)
+    .then(contents_raw => {
         var contents = contents_raw.toString();
 
         var start_marker = '<!-- begin scripts -->\n';
         var end_marker = '<!-- end scripts -->';
         var start_ind = contents.indexOf(start_marker) + start_marker.length;
         var end_ind = contents.indexOf(end_marker, start_ind);
 
+        new_script = '';
+
+        // Fallback for legacy browsers
+        new_script += `\
```

Comment:
I'd still prefer to at least have an option to disable this, but don't consider it a blocker